packages/tools/src/tools/AdvancedMagnifyViewport.ts

```diff
@@ -0,0 +1,614 @@
+      cstEvents.MOUSE_DRAG,
+      this._mouseDragCallback as EventListener
+    );
+
+    this._addNativeEventListeners(element);
+  }
+
+  private _removeEventListeners(element) {
+    eventTarget.removeEventListener(
+      cstEvents.TOOL_MODE_CHANGED,
+      this._handleToolModeChanged
+    );
+
+    element.addEventListener(
+      cstEvents.MOUSE_MOVE,
+      this._mouseDragCallback as EventListener
+    );
+
+    element.addEventListener(
+      cstEvents.MOUSE_DRAG,
+      this._mouseDragCallback as EventListener
+    );
+
+    this._removeNativeEventListeners(element);
+  }
+
+  private _initialize() {
+    const { _sourceEnabledElement: sourceEnabledElement } = this;
+    const { viewport: sourceViewport } = sourceEnabledElement;
+    const { canvas: sourceCanvas } = sourceViewport;
+    const magnifyElement = this._createViewportNode();
+
+    sourceCanvas.parentNode.appendChild(magnifyElement);
+
+    this._addEventListeners(magnifyElement);
+    this._cloneViewport(sourceViewport, magnifyElement);
+    this._enabledElement = getEnabledElement(magnifyElement);
+  }
+
+  private _syncViewportsCameras(sourceViewport, magnifyViewport) {
+    const worldPos = sourceViewport.canvasToWorld(this.position);
+
+    // Use the original viewport for the base for parallelScale
+    const parallelScale = this._convertZoomFactorToParalellScale(
+      sourceViewport,
+      magnifyViewport,
+      this.zoomFactor
+    );
+
+    const { focalPoint, position, viewPlaneNormal } =
+      magnifyViewport.getCamera();
+
+    const distance = Math.sqrt(
+      Math.pow(focalPoint[0] - position[0], 2) +
+        Math.pow(focalPoint[1] - position[1], 2) +
+        Math.pow(focalPoint[2] - position[2], 2)
+    );
+
+    const updatedFocalPoint = <Types.Point3>[
+      worldPos[0],
+      worldPos[1],
+      worldPos[2],
+    ];
+
+    const updatedPosition = <Types.Point3>[
+      updatedFocalPoint[0] + distance * viewPlaneNormal[0],
+      updatedFocalPoint[1] + distance * viewPlaneNormal[1],
+      updatedFocalPoint[2] + distance * viewPlaneNormal[2],
+    ];
+
+    magnifyViewport.setCamera({
+      parallelScale,
+      focalPoint: updatedFocalPoint,
+      position: updatedPosition,
+    });
+  }
+
+  private _syncStackViewports(
+    sourceViewport: Types.IStackViewport,
+    magnifyViewport: Types.IStackViewport
+  ) {
+    magnifyViewport.setImageIdIndex(sourceViewport.getCurrentImageIdIndex());
+  }
+
+  private _syncViewports() {
+    const { viewport: sourceViewport } = this._sourceEnabledElement;
+    const { viewport: magnifyViewport } = this._enabledElement;
+    const sourceProperties = sourceViewport.getProperties();
+
+    magnifyViewport.setProperties(sourceProperties);
+    this._syncViewportsCameras(sourceViewport, magnifyViewport);
+
+    if (this._isStackViewport(sourceViewport)) {
+      this._syncStackViewports(
+        sourceViewport as Types.IStackViewport,
+        magnifyViewport as Types.IStackViewport
+      );
+    }
```

Comment:
I expected a sync volumeViewport, but not here, not sure why not, requires a comment at least

Reply:
I was thinking about moving all code related to cloning viewports/toolGroups to `AdvancedMagnifyViewportManager` and pass two `enabledElement`s to this class. I'm not sure if that is the right place but that is an option.